posthog-core/src/index.ts

```diff
@@ -375,8 +441,8 @@ export abstract class PostHogCore {
       })
     }
 
-    // If we have flags we either return the value (true or string) or the defaultResult
-    return featureFlags[key] ?? defaultResult
+    // If we have flags we either return the value (true or string) or undefined
```

Comment:
This really confuses me 😅 

So if I call `posthog.getFeatureFlag("foo", false)` I should expect the response to be:
* `false` if the flags aren't loaded yet
* `true` if the flags are loaded and the value is true
* `undefined` if the flags are loaded but there is no flag value

This seems the opposite of what I would expect using this method.

Reply:
The way the `decide` endpoint works is that no value means the flag is set to `false`. 

This is exactly the opposite of what we want if we replace it by defaultResult, and `defaultResult` is true / some other variant string.

Reply:
... Will change! Will remove defaultResult.. in all libraries actually, and let `undefined` be the case when we fail, and let the client figure out defaults on their own.

Reply:
Actually, since all libs are complete, will just do this:

Change the 3rd bullet to
> `false` if the flags are loaded but there is no flag value

and `posthog.getFeatureFlag("foo", 'xyz')` returns `false` if the flags are loaded but there is no flag value as well.